src/lib.rs

```diff
@@ -1421,6 +1421,46 @@ pub fn build_accounts(
     account_metas
 }
 
+/// Fetch all market accounts from drift program (does not require `getProgramAccounts` RPC which is often unavailable)
+pub async fn get_market_accounts(
+    client: &RpcClient,
+) -> SdkResult<(Vec<SpotMarket>, Vec<PerpMarket>)> {
+    let state_data = client
+        .get_account_data(&state_account())
+        .await
+        .expect("state account fetch");
+    let state = State::try_deserialize(&mut state_data.as_slice()).expect("state deserializes");
+    let spot_market_pdas: Vec<Pubkey> = (0..state.number_of_spot_markets)
+        .map(|x| derive_spot_market_account(x))
+        .collect();
+    let perp_market_pdas: Vec<Pubkey> = (0..state.number_of_markets)
+        .map(|x| derive_perp_market_account(x))
+        .collect();
+
+    let (spot_markets, perp_markets) = tokio::join!(
```

Comment:
does this request cause 413s ?  i've had issues fetching more than 5 accounts at once - might test this on a couple diff free tier rpcs (alchemy, quiknode)

Reply:
tested working fine with `https://api.mainnet-beta.solana.com`, helius, rpcpool. its only once on start up